Ops/Expr/Typer.cs

```diff
@@ -37,4 +37,15 @@ public override NType Visit (NUnary unary)
          _ => Error
       };
    }
+
+   public override NType Visit (NFnCall fnCal) {
+      var types = fnCal.Params.Select (exp => exp.Accept (this)).ToArray ();
+      return fnCal.Type = (fnCal.Name.Text, types.Length) switch {
+         ("sin", 1) when (types[0] is Real or Int) => Real,
+         ("atan2", 2) when (types.All (a => a is Real or Int)) =>  Real,
+         ("length", 1) when (types[0] is NType.String) =>  Int,
+         ("random", 0) => Int,
+         _ => Error

```

Comment:
This is a good start, but will not work in the long term - in addition to these 'built-in' functions, there will also be functions the user has implemented in Pascal code. So we need a better method to support general function calls - we'll discuss in the next class.